build.gradle

```diff
@@ -1,30 +1,67 @@
 apply plugin: "cpp"
 
+def boost_home = System.getenv("BOOST_HOME")
 model {
-    binaries {
-      all {
-          linker.args "-lpthread"
+  binaries {
+    all {
+      if (buildType == buildTypes.debug) {
+        cppCompiler.args "-g"
       }
+
+      cppCompiler.args "-Wextra"
+      cppCompiler.args "-Wall"
+      cppCompiler.args "-Werror"
+
+      linker.args "-lpthread"
     }
-    buildTypes {
-        debug
-        release
+  }
+  buildTypes {
+    debug
+    release
+  }
+
+  repositories {
+    libs(PrebuiltLibraries) {
+      boost_log {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_log.a")
+        }
+      }
+      boost_thread {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_thread.a")
+        }
+      }
+      boost_system {
+        binaries.withType(StaticLibraryBinary) {
+          staticLibraryFile = file("${boost_home}/lib/libboost_system.a")
+        }
+      }
     }
-    components {
-        testrunner(NativeExecutableSpec) {
-            sources {
-                cpp {
-                    source {
-                        srcDirs += "thirdparty/gtest"
-                        srcDirs += "src"
-                        srcDirs += "test"
-                        include "**/*.cc"
-                    }
-                    exportedHeaders {
-                        srcDirs += "thirdparty/gtest"
-                    }
-                }
-            }
+  }
+
+  components {
+    testrunner(NativeExecutableSpec) {
+      sources {
+        cpp {
+          source {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
+            srcDirs += "test"
+            include "**/*.cc"
+          }
+          exportedHeaders {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
```

Comment:
@AGaliuzov, nope. Seems you're right. In general I still don't like presence of exportedHeaders for the single executable.